service_config_test.go

```diff
@@ -408,70 +407,188 @@ func (s) TestParseDefaultMethodConfig(t *testing.T) {
 
 	runParseTests(t, []parseTestCase{
 		{
-			`{
+			scjs: `{
   "methodConfig": [{
     "name": [{}],
     "waitForReady": true
   }]
 }`,
-			dc,
-			false,
+			wantSC: dc,
 		},
 		{
-			`{
+			scjs: `{
   "methodConfig": [{
     "name": [{"service": null}],
     "waitForReady": true
   }]
 }`,
-			dc,
-			false,
+			wantSC: dc,
 		},
 		{
-			`{
+			scjs: `{
   "methodConfig": [{
     "name": [{"service": ""}],
     "waitForReady": true
   }]
 }`,
-			dc,
-			false,
+			wantSC: dc,
 		},
 		{
-			`{
+			scjs: `{
   "methodConfig": [{
     "name": [{"method": "Bar"}],
     "waitForReady": true
   }]
 }`,
-			nil,
-			true,
+			wantErr: true,
 		},
 		{
-			`{
+			scjs: `{
   "methodConfig": [{
     "name": [{"service": "", "method": "Bar"}],
     "waitForReady": true
   }]
 }`,
-			nil,
-			true,
+			wantErr: true,
 		},
 	})
 }
 
 func (s) TestParseMethodConfigDuplicatedName(t *testing.T) {
 	runParseTests(t, []parseTestCase{
 		{
-			`{
+			scjs: `{
   "methodConfig": [{
     "name": [
       {"service": "foo"},
       {"service": "foo"}
     ],
     "waitForReady": true
   }]
-}`, nil, true,
+}`, wantErr: true,
+		},
+	})
+}
+
+func (s) TestParseRetryPolicy(t *testing.T) {
+	runParseTests(t, []parseTestCase{
+		{
+			name: "valid",
+			scjs: `{
+  "methodConfig": [{
+	"name": [{"service": "foo"}],
+    "retryPolicy": {
+		"maxAttempts": 2,
+		"initialBackoff": "2s",
+		"maxBackoff": "10s",
+		"backoffMultiplier": 2,
+		"retryableStatusCodes": ["UNAVAILABLE"]
+	}
+  }]
+}`,
+			wantSC: &ServiceConfig{
+				Methods: map[string]MethodConfig{
+					"/foo/": {
+						RetryPolicy: &internalserviceconfig.RetryPolicy{
+							MaxAttempts:          2,
+							InitialBackoff:       2 * time.Second,
+							MaxBackoff:           10 * time.Second,
+							BackoffMultiplier:    2,
+							RetryableStatusCodes: map[codes.Code]bool{codes.Unavailable: true},
+						},
+					},
+				},
+				lbConfig: lbConfigFor(t, "", nil),
+			},
+		},
+		{
+			name: "negative maxAttempts",
+			scjs: `{
+				"methodConfig": [{
+				  "name": [{"service": "foo"}],
+				  "retryPolicy": {
+					  "maxAttempts": -1,
+					  "initialBackoff": "2s",
+					  "maxBackoff": "10s",
+					  "backoffMultiplier": 2,
+					  "retryableStatusCodes": ["UNAVAILABLE"]
+				  }
+				}]
+			  }`, wantErr: true,
```

Comment:
Nit: Here and elsewhere, can we have `wantErr` on a separate line so that it is more readily visible.